subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/maven/MavenPomRelocationIntegrationTest.groovy

```diff
@@ -114,12 +114,12 @@ class MavenPomRelocationIntegrationTest extends AbstractHttpDependencyResolution
 
     def createBuildFileWithDependency(String groupId, String artifactId) {
         buildFile << """
+apply plugin: 'java-library'
 repositories { maven { url '${mavenHttpRepo.uri}' } }
-configurations { compile }
-dependencies { compile '${groupId}:${artifactId}:1.0' }
+dependencies { implementation '${groupId}:${artifactId}:1.0' }
 task retrieve(type: Sync) {
     into 'libs'
-    from configurations.compile
+    from configurations.compileClasspath
```

Comment:
I think it's worth adding test coverage to prove that compile and runtime scoped transitive dependencies of the relocated module end up being resolved correctly into `compileClasspath` and `runtimeClasspath`.

Reply:
Alright. This goes a bit beyond the scope of this PR, though.

Reply:
Added